### PR TITLE
chore: update stylelint to use a11y plugin

### LINKED ISSUE #388: stylelint: add a11y plugin
candidates (by a quick research):
- https://www.npmjs.com/package/@double-great/stylelint-a11y (forked from another, most likely unmaintained plugin)
- https://www.npmjs.com/package/@ronilaukkarinen/stylelint-a11y (maintained a little longer ago)
- https://www.npmjs.com/package/stylelint-accessible (promising ruleset even also for logical properties, most likely like the ones we want to bring in with https://github.com/db-ui/mono/issues/374)

Follow up on approaches by stylelint themselves like e.g. https://github.com/stylelint/stylelint/issues/6212

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #388 & https://github.com/db-ui/mono/issues/2854

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/foundations/scss/colors/_placeholder.scss [hunk 1]
```diff
@@ -155,53 +156,58 @@
...
 				--db-current-color-hover: var(
 					--db-#{$variant}-on-bg-weak-hover
 				);
 				--db-current-color-pressed: var(
 					--db-#{$variant}-on-bg-weak-pressed
 				);
+
 				color: var(--db-current-color-enabled);
 			}
 		}
```
Comment (bruno-sch) [inline packages/foundations/scss/colors/_placeholder.scss:175]: A general question about this:

What should this data-attribute be used for? The new *on-bg* classes should have the same effect. And within components, custom-properties should be used anyway - right?

File: packages/components/src/components/navigation-item/navigation-item.scss [hunk 2]
```diff
@@ -22,7 +22,14 @@
...
 	&[aria-expanded="true"] {
-		& ~ .db-sub-navigation {
+		@media screen and (prefers-reduced-motion: reduce) {
+			~ .db-sub-navigation {
+				transition: none;
+				animation: none;
+			}
+		}
+
+		~ .db-sub-navigation {
```
[the remark this answers is not in the document]
Comment (nmerget) [reply packages/components/src/components/navigation-item/navigation-item.scss]: You are right, I changed it :)

File: packages/components/src/components/drawer/drawer.scss [hunk 3]
```diff
@@ -6,6 +6,18 @@
...
 @use "./../../styles/dialog-init";
 
+%disable-animation {
+	@media screen and (prefers-reduced-motion: reduce) {
+		&:not([data-direction]),
+		&[data-direction="right"],
+		&[data-direction="left"],
+		&[data-direction="up"],
+		&[data-direction="down"] {
+			animation: none;
```
[the remark this answers is not in the document]
Comment (nmerget) [reply packages/components/src/components/drawer/drawer.scss]: See: https://github.com/db-ui/mono/pull/2677#discussion_r1658337069

File: .stylelintrc.json [hunk 4]
```diff
@@ -1,25 +1,24 @@
 {
 	"extends": [
-		"stylelint-config-sass-guidelines",
-		"stylelint-config-prettier"
```
Comment (mfranzke) [inline .stylelintrc.json:4]: why have you removed `stylelint-config-prettier`?

File: .stylelintrc.json [hunk 5]
```diff
@@ -1,25 +1,24 @@
...
 		"csstools/use-logical": ["always", { "except": ["float"] }],
-		"order/properties-alphabetical-order": null,
-		"scss/operator-no-newline-after": null,
-		"declaration-empty-line-before": null,
-		"selector-no-qualifying-type": null,
-		"selector-class-pattern": null,
-		"scss/dollar-variable-pattern": null,
-		"scss/selector-no-redundant-nesting-selector": null,
 		"max-nesting-depth": null,
+		"selector-type-no-unknown": null,
```
Comment (mfranzke) [inline .stylelintrc.json:13]:
```suggestion
		"selector-type-no-unknown": [true, { "ignoreTypes": ["/^db/"] }],
```

File: .stylelintrc.json [hunk 6]
```diff
@@ -1,25 +1,24 @@
...
+		"no-descending-specificity": null,
```
Comment (mfranzke) [inline .stylelintrc.json:14]: This rule seems to make sense to me in general – is there a misinterpretation or even some problem that occurs our of [SCSS currently handling the input vs. the output order of declarations](https://sass-lang.com/documentation/breaking-changes/mixed-decls/)?

File: packages/components/src/components/accordion-item/accordion-item.scss [hunk 7]
```diff
@@ -65,9 +64,11 @@ $db-accordion-item-border-radius: variables.$db-border-radius-sm;
...
 
 	&[open] {
-		summary + div {
-			animation: accordion-open
-				variables.$db-transition-straight-emotional normal both;
+		@media screen and (prefers-reduced-motion: no-preference) {
+			summary + div {
+				animation: accordion-open
+					variables.$db-transition-straight-emotional normal both;
+			}
```
Comment (mfranzke) [inline packages/components/src/components/accordion-item/accordion-item.scss:71]:
```suggestion
		summary + div {
			@media screen and (prefers-reduced-motion: no-preference) {
				animation: accordion-open
					variables.$db-transition-straight-emotional normal both;
			}
```

File: packages/components/src/components/accordion-item/accordion-item.scss [hunk 8]
```diff
@@ -65,9 +64,11 @@ $db-accordion-item-border-radius: variables.$db-border-radius-sm;
 	}
 
 	&[open] {
-		summary + div {
-			animation: accordion-open
-				variables.$db-transition-straight-emotional normal both;
+		@media screen and (prefers-reduced-motion: no-preference) {
+			summary + div {
+				animation: accordion-open
```
Comment (mfranzke) [inline packages/components/src/components/accordion-item/accordion-item.scss:69]: I'm wondering whether we could more easily add these wrapping declarations via a postcss plugin generally, but haven't found one quickly.

File: packages/components/src/components/navigation-item/navigation-item.scss [hunk 9]
```diff
@@ -23,10 +23,16 @@
...
-		& ~ .db-sub-navigation {
+		@media screen and (prefers-reduced-motion: no-preference) {
+			~ .db-sub-navigation {
+				transition: visibility 0ms linear 0ms;
+				animation: show-right-to-left
+					variables.$db-transition-straight-show;
+			}
+		}
+
+		~ .db-sub-navigation {
```
Comment (mfranzke) [inline packages/components/src/components/navigation-item/navigation-item.scss:34]:
```suggestion
		~ .db-sub-navigation {
		@media screen and (prefers-reduced-motion: no-preference) {
			transition: visibility 0ms linear 0ms;
			animation: show-right-to-left
				variables.$db-transition-straight-show;
		}
```

Simplification

File: packages/components/src/styles/_popover-component.scss [hunk 10]
```diff
@@ -12,25 +12,33 @@ $popover-gap-inset: calc(-1 * #{$popover-gap-size});
...
+	&[data-outside-vy="bottom"]:not(
+			[data-outside-vx],
+			[data-placement$="end"],
+			[data-placement$="start"]
+		),
+	&[data-outside-vy="top"]:not(
+			[data-outside-vx],
+			[data-placement$="end"],
+			[data-placement$="start"]
+		) {
```
Comment (mfranzke) [inline packages/components/src/styles/_popover-component.scss:24]:
```suggestion
	&:is([data-outside-vy="bottom"], [data-outside-vy="top"]):not(
			[data-outside-vx],
			[data-placement$="end"],
			[data-placement$="start"]
		) {
```

File: packages/foundations/scss/colors/_placeholder.scss [hunk 11]
```diff
@@ -155,53 +156,58 @@
 		@include bg-types() {
 			@extend %set-current-colors;
 
-			&[data-emphasis="weak"] {
+			[data-emphasis="weak"] {
```
Comment (mfranzke) [inline packages/foundations/scss/colors/_placeholder.scss:159]: while removing the `&` characters at the beginning at other places in the code, this change results in that elements with the attribute `[data-emphasis="weak"]` *within* elements that are extended with the `%db-#{$variant}-bg` placeholder would get selected. I assume, that this `&` character was removed by mistake.

Comment (nmerget) [reply packages/foundations/scss/colors/_placeholder.scss]: it's not relevant anymore

File: packages/components/src/components/accordion-item/accordion-item.scss (hunk 8 above)
Comment (nmerget) [reply packages/components/src/components/accordion-item/accordion-item.scss:69]: The stylelint plugin adds those with the `--fix` option

File: .stylelintrc.json [hunk 12]
```diff
@@ -1,25 +1,24 @@
...
-		"scss/at-mixin-pattern": null,
-		"scss/at-import-partial-extension": null,
-		"selector-max-id": null
+		"selector-no-qualifying-type": [
+			true,
+			{
+				"ignore": ["attribute"]
+			}
+		],
+		"number-max-precision": null
```
[the remark this answers is not in the document]
Comment (nmerget) [reply .stylelintrc.json]: This decleration just adds some false positives for `animation` and `transition`, why should we use it?

File: .stylelintrc.json (hunk 6 above)
Comment (nmerget) [reply .stylelintrc.json:14]: In our case it isn't working properly in SCSS...
Maybe we could write the `scss` files differently to fit the rule, but in my opinion, it's just a restriction. I like to group my rules and don't care where they are written in the file.

File: .stylelintrc.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (mfranzke) [reply .stylelintrc.json]: because it's a cool new rule.

File: packages/components/src/components/switch/switch.scss [hunk 13]
```diff
@@ -208,7 +206,8 @@ $checked-active-transition-size: calc(
 		}
 	}
 
-	& > input {
+	/* stylelint-disable */
+	> input {
```
Comment (mfranzke) [inline packages/components/src/components/switch/switch.scss:210]: @nmerget what's the reason not to put those into the other `> input` section of this file, like e.g. at line 130 ?

**Review by mfranzke — DISMISSED** [review]

File: packages/components/src/components/switch/switch.scss (hunk 13 above)
Comment (nmerget) [reply packages/components/src/components/switch/switch.scss:210]: If we set `data-aid-icon` or `data-aid-icon-after` it should overwrite the default icons. Therefore I put it at the end of the file

Comment (mfranzke) [reply packages/components/src/components/switch/switch.scss:210]: Okay, I've added this as a comment for us to remember.

**Review by mfranzke — APPROVED** [review]

**Pull request merged**